.circleci/config.yml

```diff
@@ -20,4 +22,19 @@ jobs:
               echo "Awesome! GCS integration tests are enabled."
             fi
       - run: make format
-      - run: make test
\ No newline at end of file
+      - run: make test
+      - run: make docker
+      - run:
+          name: Run tests on generated Docker image
+          command: |
+            docker run improbable/thanos --help
+            docker run improbable/thanos sidecar --help
+            docker run improbable/thanos store --help
+            docker run improbable/thanos query --help
+      - deploy:
+          name: Push Docker image
+          command: |
+            docker login -u="${DOCKER_USERNAME}" -p="${DOCKER_PASSWORD}"
```

Comment:
Let's move this login into below `if` as well! This is critical, because env vars will be not evaluated for forked PRs. (and is only needed for master)

Reply:
Yup, my bad.